vault/plugin_catalog.go

```diff
@@ -1037,6 +1041,33 @@ func (c *PluginCatalog) List(ctx context.Context, pluginType consts.PluginType)
 	return retList, nil
 }
 
+// ListPluginsWithRuntime lists the plugins that are registered with a given runtime
+func (c *PluginCatalog) ListPluginsWithRuntime(ctx context.Context, runtime string) ([]string, error) {
```

Comment:
I looked into earlier in the call stack that constructs `PluginDetails`. Looks like `oci_image` and `runtime` don't need to be added here since`detailed` only contains the versioned plugins https://github.com/hashicorp/vault/blob/main/vault/logical_system.go#L463-L474.
I removed them in [this commit]([remove oci_image and runtime from PluginDetails since versioned plugi…](https://github.com/hashicorp/vault/pull/22744/commits/4a3ad9c114de594a60766356a4a77b71b1a066b1)).